proposals/semantic_schema/README.md

```diff
@@ -0,0 +1,154 @@
+
+- `Kind` - Must be pre-defined. Cannot vary over time. Better suited for filtering or interacting with objects on the Stage, not for rendering or label data.
+
+### Why a list?
+
+What an object "is" becomes very tricky with the nature of human language. Is the object an `automobile`, a `vehicle`, or the more informal `car`?  
+This is especially true when considering multiple languages: `car`, `Auto`, `voiture`, `bil`, `coche`, etc.
+
+Objects can also have more detailed semantics depending on their context.
+
+Examples:
+ -  `['vehicle', 'emergency vehicle', 'ambulance']`
+ -  `['chair', 'office furniture', 'executive chair']`
+
+By having a list, semantics on a prim can be additive, increasing the understanding of what the object is for more people and use cases, allowing filtering for the terms that are applicable to the user.
+
+
+### Encoding semantic "type"
+
+Within the machine learning community, 'class' is a very common moniker for defining a category for semantic labels, but there could be more "types" of semantic labels. An object's `class` could be `vehicle`, its `sub-class` `ambulance`, and it's material semantics `metal` or `steel` for certain parts, etc.
+
+For this we took inspiration from existing USD Schemas such as `CoordSysAPI` and light linking that encode meaning in the instance name of the schema object.
+
+Preferring:
+
+```
+token[] semantics:class:labels = ["animal", "bird", "penguin"]
+```
+
+instead of multiple instances on the same prim just to contain all of the information for all possible `class` labels
+
+```
+token semantics:Semantics_aDa0:semanticType = "class"
+token semantics:Semantics_aDa0:semanticLabel = "animal"
+
+token semantics:Semantics_d282:semanticType = "class"
+token semantics:Semantics_d282:semanticLabel = "bird"
+
+token semantics:Semantics_c0nr:semanticType = "class"
+token semantics:Semantics_c0nr:semanticLabel = "penguin"
+```
+
+Or requiring string parsing operations
+
+```
+string[] semantics:instance_name:labels = ["class:animal", "class:bird", "class:penguin"]
+```
+
+The proposed method has better performance as the number of labels of the same type increase on a prim to dozens, hundreds, or thousands of labels.
+
+### Tokens vs Strings
+
+Consideration was taken for the datatype being tokens or strings.  Considering that semantic labels should be more read-intensive rarely (if ever) written/updated, tokens are thought to be the better representation.
+
+Some performance testing was done, and in some cases StringArray performed better than TokenArray in Python, the difference was negligible and could be attributed to the overhead of converting tokens to native Python strings.
+
+Tokes might also facilitate future expansion use of `allowedTokens` for contraining semantic labels, but that will not be a part of this proposal at this time.
+
```

Comment:
@dennis-lynch-nv , I think I recall from the ASWF discussion that a key aspect of this schema is a namespace inheritance behavior that, within a particular multi-apply "instance", the resolved value of `semantics:myConcept:labels` is the **aggregation as a list** (or is it a unique set?) of all the labels of all ancestors of the prim being queried.  That is also one of the key selling points for me as to why we want to formalize a schema here.  If this is accurate, can you please add a section that motivates and describes the behavior?

Thank you!

Reply:
@spiffmon - We've discussed this some internally and see your point.  We should probably do a unique set (list of Tokens?) upon traversal/collapse of the semantic hierarchy.

```
└── xform - car
    ├── wheel
    ├── door
    │   ├── window(glass)
    │   └── handle
    └── seat
```
- Query the door prim, get `semantics:myConcept:labels = ['door']`
- Query door prim for "Parents" or "Ancestors":   `semantics:myConcept:labels = ['door', 'car']`
- Query handle prim for "Parents" or "Ancestors":   `semantics:myConcept:labels = ['handle', 'door', 'car']`
- Query for "Children" or "Descendents" from door prim:  `semantics:myConcept:labels = ['door', 'handle']`
- Only semantics of the same "type"/instance name (ex: `myConcept`) are combined.  If the car prim had `semantics:otherConcept:labels = ['Incredibile']` it would not be added to the list of `semantics:myConcept:labels` when queried from the `myConcept` instance on the door or handle

What do you think?

As far as nomenclature- I wasn't sure if we would _want_ to keep the same terms that `Usd` already uses for prims (`GetChildren()`, `GetChild()`, `GetParent()`, etc.) or would we want to use different terminology to avoid confusion about Prim hierarchy traversal vs semantic hierarchy traversal

Reply:
"Ancestors"-only is fine, I was only including "Descendents" for reasons of symmetry, but there are not a lot/any known reasons to regularly get semantic descendants, so if it is also a performance issue, then we can leave that out.

Reply:
I'm unclear if the patch visible above is the resolution of this conversation, or not?

Reply:
Maybe @spiffmon could comment if this is sufficient or he has something else in mind?

Reply:
Added one request for clarification, but otherwise the additions look good to me!